dubbo-common/src/main/java/org/apache/dubbo/common/utils/NetUtils.java

```diff
@@ -231,21 +216,31 @@ public static InetAddress getLocalAddress() {
         return localAddress;
     }
 
+    private static Optional<InetAddress> toValidAddress(InetAddress address) {

```

Comment:
`Optional` requires some costs regarding preformance, I'm not sure we should encourage it or not in Dubbo.

Reply:
This is yet to be decided. Personally think that you can use the judgment of null. As we have seen, most frameworks internally support 1.8, but rarely use optinal.

Reply:
As we discussed in mail list, the community encourage us to use `Optional` :)